### PR TITLE
Simplify field reordering, preserve event handlers on unchanged DOM.

### PR DESCRIPTION
**Pull request by jtwalters**
- Simplify field reordering, preserve event handlers on unchanged DOM.
- Drop jQuery 1.3 support.
- Suggest bumping version to 2.0.0.

### CODE REVIEW

**Comment by jtwalters** [comment]
@iamEAP this changeset is related to better supporting float labels. It simplifies your code and preserves DOM without using `.clone()`, which also preserves event handlers on DOM (at least DOM which is unmodified by this plugin).

**Comment by iamEAP** [comment]
Thanks @jtwalters!  Is the jQuery 1.3 drop required for the `orderFields` refactor, or can we get away with pulling that into a separate PR?

**Comment by jtwalters** [comment]
Hmm, well, I'm using `.detach()` instead of `.remove()` which isn't available in jQuery 1.3. This change might not be required for our redesign related work, but it does seem to be desirable to avoid losing all the event handlers that might be attached (?) It's tough to say whether this is useful for folks.

**Comment by iamEAP** [comment]
Ah, gotcha.  This definitely seems useful, especially for folks using this plugin in the context of single-page apps, Angular, etc.

I just want as many people as possible to take advantage of this bugfix.  Bumping to 2.0.x complicates downstream releases (e.g. the Drupal module that integrates this library would have to also bump a version).

How about for a 1.1.x release, adding a `.detach()` implementation [like this](http://stackoverflow.com/a/7026902) with a `@todo` and PR to remove it (and the other 1.3 code)?

**Comment by jtwalters** [comment]
Ok, that seems reasonable to use a custom `.detach()` which is compatible with jQuery 1.3. I'll work on a backward compatible PR that only requires a minor version bump.
